aproximation_rules.py

```diff
@@ -0,0 +1,58 @@
+import numpy
+
+def mid_rule(n, interval, my_func):
+    '''Approximates the area between a given curve f(x) and the line x=0 within a closed interval [a,b] using the Midpiont Rule '''
+
+    delta_x = (interval[1] - interval[0])/n
+
+    endpoints = []
+    for i in numpy.arange(interval[0]+delta_x, interval[1]+delta_x, delta_x):
+        endpoints.append( 1./2 * ( (i-delta_x) + i))
+
+    endpoints = my_func(endpoints)
+    result = sum(endpoints)

```

Comment:
if endpoints is still a numpy array here consider using np.sum(endpoint), it's faster